index.js

```diff
@@ -112,90 +130,107 @@ module.exports = {
+        });
+
+        const redNodes = this._redNodes;
+        redNodes.init({
+            events: this._events,
+            settings: settings,
+            storage:storage,
+            log:this._log
+        });
+        redNodes.registerType("helper", function (n) {
+            redNodes.createNode(this, n);
+        });
+
         if (Array.isArray(testNode)) {
-            for (i = 0; i < testNode.length; i++) {
-                testNode[i](red);
-            }
+            testNode.forEach(fn => {
+                fn(red);
+            });
         } else {
             testNode(red);
         }
-        redNodes.loadFlows().then(function() {
-            redNodes.startFlows();
-            should.deepEqual(testFlow, redNodes.getFlows().flows);
-            cb();
-        });
-    },
 
-    unload: function() {
+        redNodes.loadFlows()
+            .then(() => {
+                redNodes.startFlows();
+                should.deepEqual(testFlow, redNodes.getFlows().flows);
+                cb();
+            });
+    }
+
+    unload() {
         // TODO: any other state to remove between tests?
-        redNodes.clearRegistry();
-        logSpy.restore();
-        // internal API
-        context.clean({allNodes:[]});
-        return redNodes.stopFlows();
-    },
+        this._redNodes.clearRegistry();
+        this._logSpy.restore();
+        this._sandbox.reset();
 
-    getNode: function(id) {
-        return redNodes.getNode(id);
-    },
+        // internal API
+        this._context.clean({allNodes:[]});
+        return this._redNodes.stopFlows();
+    }
 
-    credentials: credentials,
+    /**
+     * Returns a Node by id.
+     * @param {string} id - Node ID
+     * @returns {Node}
+     */
+    getNode(id) {
+        return this._redNodes.getNode(id);
+    }
 
-    clearFlows: function() {
-        return redNodes.stopFlows();
-    },
+    clearFlows() {
+        return this._redNodes.stopFlows();
+    }
 
-    request: function() {
-        return request(RED.httpAdmin);
-    },
+    request() {
+        return request(this._RED.httpAdmin);
+    }
 
-    startServer: function(done) {
-        server = stoppable(http.createServer(function(req,res) { app(req,res); }), 0);
+    startServer(done) {
+        this._app = express();
+        const server = stoppable(http.createServer((req,res) => {
+            this._app(req,res);
+        }), 0);
 
-        RED.init(server, {
+        this._RED.init(server, {
             SKIP_BUILD_CHECK: true,
             logging:{console:{level:'off'}}
         });
-        server.listen(listenPort, address);
-        server.on('listening', function() {
-            port = server.address().port;
-            url = 'http://' + address + ':' + port;
+        server.listen(this._listenPort, this._address);
+        server.on('listening', () => {
+            this._port = server.address().port;
+            this.url = `http://${this._address}:${this._port}`;
```

Comment:
needs to be:

this.url = () => `http://${this._address}:${this._port}`;

since url was a function before, otherwise core debug node test fails

Reply:
my bad; good catch

Reply:
I suggest also that you move this assignment to a class method so that the object's interface is more obvious.  i.e. use

 `function url() { return `http://${this._address}:${this._port}`;}`